File: trunk/p4o/component/biz/Class_biz_squads.pas
unit Class_biz_squads;

interface

uses
  Class_db_squads,
  kix;

type
  TClass_biz_squads = class
  published
    function Bind
      (
      partial_code: string;
      target: system.object
      )
      : boolean;
    procedure BindDirectToListControl
      (
      target: system.object;
      unselected_literal: string = '-- Squad --';
      selected_value: string = EMPTY
      );
    function Delete(code: string): boolean;
    function Get
      (
      code: string;
      out description: string;
      out unit_id: string
      )
      : boolean;
    procedure &Set
      (
      code: string;
      description: string;
      unit_id: string
      );
  public
    constructor Create;
  strict private
    db_squads: TClass_db_squads;
  end;

implementation

constructor TClass_biz_squads.Create;
begin
  inherited Create;
  // TODO: Add any constructor code here
  db_squads := TClass_db_squads.Create;
end;

function TClass_biz_squads.Bind
  (
  partial_code: string;
  target: system.object
  )
  : boolean;
begin
  Bind := db_squads.Bind(partial_code,target);
end;

procedure TClass_biz_squads.BindDirectToListControl
  (
  target: system.object;
  unselected_literal: string = '-- Squad --';
  selected_value: string = EMPTY
  );
begin
  db_squads.BindDirectToListControl(target,unselected_literal,selected_value);
end;

function TClass_biz_squads.Delete(code: string): boolean;
begin
  Delete := db_squads.Delete(code);
end;

function TClass_biz_squads.Get
  (
  code: string;
  out description: string;
  out unit_id: string
  )
  : boolean;
begin
  //
  Get := db_squads.Get
    (
    code,
    description,
    unit_id
    );
  //
end;

procedure TClass_biz_squads.&Set
  (
  code: string;
  description: string;
  unit_id: string
  );
begin
  //
  db_squads.&Set
    (
    code,
    description,
    unit_id
    );
  //
end;

end.

File: component/biz/Class_biz_squads.cs
using kix;
using System;
using Class_db_squads;
namespace Class_biz_squads
{
    public class TClass_biz_squads
    {
        private TClass_db_squads db_squads = null;
        //Constructor  Create()
        public TClass_biz_squads() : base()
        {
            // TODO: Add any constructor code here
            db_squads = new TClass_db_squads();
        }
        public bool Bind(string partial_code, object target)
        {
            bool result;
            result = db_squads.Bind(partial_code, target);
            return result;
        }

        public void BindDirectToListControl(object target, string unselected_literal, string selected_value)
        {
            db_squads.BindDirectToListControl(target, unselected_literal, selected_value);
        }

        public void BindDirectToListControl(object target)
        {
            BindDirectToListControl(target, "-- Squad --");
        }

        public void BindDirectToListControl(object target, string unselected_literal)
        {
            BindDirectToListControl(target, unselected_literal, k.EMPTY);
        }

        public bool Delete(string code)
        {
            bool result;
            result = db_squads.Delete(code);
            return result;
        }

        public bool Get(string code, out string description, out string unit_id)
        {
            bool result;
            result = db_squads.Get(code, out description, out unit_id);

            return result;
        }

        public void Set(string code, string description, string unit_id)
        {
            db_squads.Set(code, description, unit_id);

        }

    } // end TClass_biz_squads

}

namespace Class_biz_squads.Units
{
    public class Class_biz_squads
    {
    } // end Class_biz_squads

}

